src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2232,24 +2237,112 @@ private void VisitBinaryOperatorChildren(BoundBinaryOperator node)
             stack.Free();
         }
 
+#nullable enable
         protected virtual void VisitBinaryOperatorChildren(ArrayBuilder<BoundBinaryOperator> stack)
         {
             var binary = stack.Pop();
-            VisitRvalue(binary.Left);
 
-            while (true)
+            // Only the leftmost operator of a left-associative binary operator chain can learn from a conditional access on the left
+            // For simplicity, we just special case it here.
+            // For example, `a?.b(out x) == true` has a conditional access on the left of the operator,
+            // but `expr == a?.b(out x) == true` has a conditional access on the right of the operator
+            if (VisitPossibleConditionalAccess(binary.Left, out var stateWhenNotNull)
+                && canLearnFromOperator(binary)
+                && isKnownNullOrNotNull(binary.Right))
             {
                 VisitRvalue(binary.Right);
+                Meet(ref stateWhenNotNull, ref State);
+                var isNullConstant = binary.Right.ConstantValue?.IsNull == true;
+                SetConditionalState(isNullConstant == isEquals(binary)
+                    ? (State, stateWhenNotNull)
+                    : (stateWhenNotNull, State));
+
+                if (stack.Count == 0)
+                {
+                    return;
+                }
+
+                binary = stack.Pop();
+            }
+
+            while (true)
+            {
+                if (!canLearnFromOperator(binary)
+                    || !learnFromOperator(binary))
+                {
+                    Unsplit();
+                    Visit(binary.Right);
+                }
 
                 if (stack.Count == 0)
                 {
                     break;
                 }
 
-                Unsplit(); // VisitRvalue does this
                 binary = stack.Pop();
             }
+
+            static bool canLearnFromOperator(BoundBinaryOperator binary)
+            {
+                var kind = binary.OperatorKind;
+                return kind.Operator() is BinaryOperatorKind.Equal or BinaryOperatorKind.NotEqual
+                    && (!kind.IsUserDefined() || kind.IsLifted());
+            }
+
+            static bool isKnownNullOrNotNull(BoundExpression expr)
+            {
+                return expr.ConstantValue is object
+                    || (expr is BoundConversion
+                        {
+                            ConversionKind: ConversionKind.ExplicitNullable or ConversionKind.ImplicitNullable,
+                        } conv && conv.Operand.Type!.IsNonNullableValueType());
```

Comment:
```suggestion
                    || (expr is BoundConversion { ConversionKind: ConversionKind.ExplicitNullable or ConversionKind.ImplicitNullable } conv
                        && conv.Operand.Type!.IsNonNullableValueType());
```

Blah, it doesn't want the brace indented like this. Trying different formatting.

`src\Compilers\CSharp\Portable\FlowAnalysis\AbstractFlowPass.cs(2296,21): error IDE0055: (NETCORE_ENGINEERING_TELEMETRY=Build) Fix formatting`